src/Compilers/CSharp/Test/Symbol/Symbols/Source/MethodTests.cs

```diff
@@ -203,6 +204,84 @@ public partial class A {
         }
 
         [Fact]
+        public void PartialExtractSyntaxLocation_DeclBeforeDef()
+        {
+            var text =
+@"public partial class A {
+  partial void M();
+}
+public partial class A {
+  partial void M() {}
+}
+";
+            var comp = CreateStandardCompilation(text);
+            var global = comp.GlobalNamespace;
+            var a = global.GetTypeMembers("A", 0).Single();
+            var m = a.GetMembers("M");
+            Assert.Equal(1, m.Length);
+            var returnSyntax = ((MethodSymbol) m.First()).ExtractReturnTypeSyntax();
+            var newLineCount = Environment.NewLine.Length * 1;
+            Assert.Equal(returnSyntax.Location.SourceSpan, TextSpan.FromBounds(34 + newLineCount, 38 + newLineCount));
```

Comment:
>Assert.Equal(returnSyntax.Location.SourceSpan, TextSpan.FromBounds(34 + newLineCount, 38 + newLineCount)); [](start = 12, length = 106)

I think it would be better to find the node that you expect directly from the syntax tree. This way I cannot really tell if your expectations are correct. Also, I believe expected value should be first, otherwise failure message is confusing.